src/SGFTree.cpp

```diff
@@ -181,10 +185,30 @@ void SGFTree::populate_states() {
         }
     }
 
+    // time
+    it = m_properties.find("TM");
+    if (it != end(m_properties)) {
+        const auto maintime = it->second;
+        it = m_properties.find("OT");
+        const auto byoyomi = (it != end(m_properties)) ? it->second : "";
+        it = m_properties.find("BL");
+        const auto black_time_left = (it != end(m_properties)) ? it->second : "";
+        it = m_properties.find("WL");
+        const auto white_time_left = (it != end(m_properties)) ? it->second : "";
+        it = m_properties.find("OB");
+        const auto black_moves_left = (it != end(m_properties)) ? it->second : "";
+        it = m_properties.find("OW");
+        const auto white_moves_left = (it != end(m_properties)) ? it->second : "";
+        m_timecontrol_ptr = std::make_shared<TimeControl>();
+        m_timecontrol_ptr->set_from_text_sgf(maintime, byoyomi,
```

Comment:
Does this need some sanity checks in case parts are missing or inconsistent?

Reply:
Guess this could be a shared_ptr<TimeControl> factory as it's the only caller of set_from_text_sgf, i.e. make_from_text_sgf.

Reply:
> Does this need some sanity checks in case parts are missing or inconsistent?

If we're only trying to be self-consistent and lossless then no. Note that there is no mention of leading time settings from an sgf in the GTP spec so the controller should be sending time settings separately... which sounds like a problem if we override previously sent time settings :(

I think the only case that could be inconsistent is if there is no byo-yomi but the sgf has remaining byo-yomi stones recorded. Other than there there is the case in which the byo-yomi syntax is unrecognised (which could lead to the preceding case). If the controller sends the time controls after then it would be bad to not load the sgf.

Hmm. Thoughts?